modules/local/fastp/main.nf

```diff
@@ -19,13 +24,23 @@ process FASTP {
         */
         '''
         # Define paths and subcommands
-        o1=!{sample}_fastp_1.fastq.gz
-        o2=!{sample}_fastp_2.fastq.gz
         of=!{sample}_fastp_failed.fastq.gz
         oj=!{sample}_fastp.json
         oh=!{sample}_fastp.html
         ad=!{adapters}
-        io="--in1 !{reads[0]} --in2 !{reads[1]} --out1 ${o1} --out2 ${o2} --failed_out ${of} --html ${oh} --json ${oj} --adapter_fasta ${ad}"
+        if [ $(echo "!{reads}" | wc -w) -eq 2 ]; then
```

Comment:
For some processes, you split the process into two (and call is SINGLE vs PAIRED), whereas in others, you use if statements with in the process (like here for example). What's been your criteria for choosing one over the other? I think we should standardize this if possible.

Reply:
Roughly depends on how large the changes are. Sometimes, Will had a preference for putting it into the process, sometimes not. Here I actually now switched to having fastp have a single-end and paired-end version.

Reply:
Chatted with Will, going forward, make sure everything is separate, i.e. have a single and a paired process. Thanks for doing that here!